test/helpers/policygen/policygen.go

```diff
@@ -0,0 +1,133 @@
+package policygen
+
+var policiesTestSuite = PolicyTestSuite{
+	l3Checks: []PolicyTest{
+		PolicyTest{
+			name:  "No Policy",
+			kind:  ingress,
+			tests: ConnResultAllOK,
+			template: map[string]string{
+				"FromEndpoints": `[{}]`,
+			},
+		},
+		PolicyTest{
+			name:  "Ingress Label",
+			kind:  ingress,
+			tests: ConnResultAllOK,
+			template: map[string]string{
+				"FromEndpoints": `[{"matchLabels": { "id": "{{.SrcPod}}"}}]`,
+			},
+		},
+		PolicyTest{
+			name:  "Ingress Label Invalid",
+			kind:  ingress,
+			tests: ConnResultAllTimeout,
+			template: map[string]string{
+				"FromEndpoints": `[{"matchLabels": { "id": "{{.SrcPod}}Invalid"}}]`,
+			},
+		},
+	},
+	l4Checks: []PolicyTest{
+		PolicyTest{
+			name:     "No Policy",
+			kind:     ingress,
+			tests:    ConnResultAllOK,
+			template: map[string]string{},
+		},
+		PolicyTest{
+			name:  "Ingress Port 80 No protocol",
+			kind:  ingress,
+			tests: ConnResultOnlyHTTP,
+			template: map[string]string{
+				"Ports": `[{"port": "80"}]`,
+			},
+		},
+		// PolicyTest{
+		// 	name: "Ingress proto TCP",
+		// 	kind: ingress,
+		// 	tests: ConnTestSpec{
+		// 		Ping:        ResultTimeout,
+		// 		HTTP:        ResultOK,
+		// 		UDP:         ResultTimeout,
+		// 		HTTPPrivate: ResultOK,
+		// 	},
+		// 	template: map[string]string{
+		// 		"Ports": `[{"protocol": "TCP"}]`,
+		// 	},
+		// },
+		// PolicyTest{
+		// 	name: "Ingress proto ICMP",
+		// 	kind: ingress,
+		// 	tests: ConnTestSpec{
+		// 		Ping:        ResultTimeout,
+		// 		HTTP:        ResultOK,
+		// 		UDP:         ResultTimeout,
+		// 		HTTPPrivate: ResultOK,
+		// 	},
+		// 	template: map[string]string{
+		// 		"Ports": `[{"protocol": "icmp"}]`,
+		// 	},
+		// },
```

Comment:
I'm not sure why this is commented, but it would be good to either uncomment&fix, or open an issue and add a line above the comment that links to the GH issue so we can know why this commented code exists, and find out more.